pulp_container/tests/functional/api/test_sync.py

```diff
@@ -1,231 +1,147 @@
 """Tests that sync container plugin repositories."""
-from pulp_smash import config
-from pulp_smash.pulp3.bindings import (
-    delete_orphans,
-    monitor_task,
-    PulpTaskError,
-    PulpTestCase,
-)
-from pulp_smash.pulp3.utils import gen_repo
-from pulp_container.tests.functional.utils import (
-    gen_container_client,
-    gen_container_remote,
-)
+import pytest
+from pulpcore.tests.functional import PulpTaskError
+
+from pulp_container.tests.functional.utils import gen_container_remote
 from pulp_container.tests.functional.constants import PULP_FIXTURE_1
 
-from pulpcore.client.pulp_container import (
-    ContainerContainerRepository,
-    ContainerContainerRemote,
-    ContainerRepositorySyncURL,
-    ContentTagsApi,
-    RepositoriesContainerApi,
-    RepositoriesContainerVersionsApi,
-    RemotesContainerApi,
-)
+from pulpcore.client.pulp_container import ContainerContainerRemote
 
 
-class BasicSyncTestCase(PulpTestCase):
-    """Sync repositories with the container plugin."""
+@pytest.fixture
+def synced_repository_with_filtered_tags(
+    container_repo, container_remote_api, gen_object_with_cleanup, sync
```

Comment:
I don't think it is intuitive that this fixture changes another fixture. I'd prefer to use a factory inside this one.
```suggestion
    container_repository_factory, container_remote_api, gen_object_with_cleanup, sync
```

A factory for remotes may be just as handy.

Reply:
I think I didn't get your point here @mdellweg. Isn't the `synced_repository_with_filtered_tags` a factory? Think I may missing something of your comment.

Reply:
I'm talking about the `container_repo` that is not a factory, but used by this factory. If you this factory twice, it would change only the one repository twice.